src/textual/binding.py

```diff
@@ -193,6 +236,71 @@ def merge(cls, bindings: Iterable[BindingsMap]) -> BindingsMap:
                 keys.setdefault(key, []).extend(key_bindings)
         return BindingsMap.from_keys(keys)
 
+    def apply_keymap(self, keymap: Keymap) -> KeymapApplyResult:
```

Comment:
Is this a destructive operation? i.e. can I update the keymap on the fly if I wanted to?

Just thinking about the possibility of a keymap editor in the future, or the ability to load keymaps on the fly.

Reply:
It's destructive in that it will completely replace the _key_ associated with a binding in the bindings map (although still retrievable via the default `BINDINGS` list). However, the binding would still maintain it's ID, so could be re-assigned to any key at any time.

Updating on the fly should work as it stands, as the keymap is just "overlayed" over the bindings on-demand, but I suspect you'd need to force a refresh of the key panel and footer.